usr/lib/cca_stdll/cca_specific.c

```diff
@@ -299,22 +295,15 @@ static CK_RV cca_key_gen(STDLL_TokData_t *tokdata, TEMPLATE *tmpl,
 static CK_RV init_cca_adapter_lock(STDLL_TokData_t *tokdata)
 {
     struct cca_private_data *cca_private = tokdata->private_data;
-    pthread_rwlockattr_t attr;
+    pthread_rwlockattr_t attr = NULL;
     unsigned long cnt;
+    int rc;
 
-    if (pthread_rwlockattr_init(&attr) != 0) {
-        TRACE_ERROR("pthread_rwlockattr_init failed\n");
-        OCK_SYSLOG(LOG_ERR, "%s: Failed to initialize the CCA adapter lock\n",
-                   __func__);
-        return CKR_CANT_LOCK;
-    }
-
-    int rc = pthread_rwlock_init(&cca_private->min_card_version_rwlock, &attr);
+    rc = pthread_rwlock_init(&cca_private->min_card_version_rwlock, NULL);
```

Comment:
This function is to initialize the global `cca_adapter_rwlock` only. It should not also initialize `cca_private->min_card_version_rwlock`. 

 `cca_private->min_card_version_rwlock` is initialized in `token_specific_init()` but unfortunately it is erroneously ifdefed out with NO_PEKY:

```
#ifndef NO_PKEY
    cca_private->msa_level = get_msa_level();
    TRACE_INFO("MSA level = %i\n", cca_private->msa_level);

    if (pthread_rwlock_init(&cca_private->min_card_version_rwlock, NULL) != 0) {
        TRACE_ERROR("Initializing the min_card_version RW-Lock failed\n");
        rc = CKR_CANT_LOCK;
        goto error;
    }
```

So this needs to be fixed in `token_specific_init()` as follows:

```
#ifndef NO_PKEY
    cca_private->msa_level = get_msa_level();
    TRACE_INFO("MSA level = %i\n", cca_private->msa_level);
#endif

    if (pthread_rwlock_init(&cca_private->min_card_version_rwlock, NULL) != 0) {
        TRACE_ERROR("Initializing the min_card_version RW-Lock failed\n");
        rc = CKR_CANT_LOCK;
        goto error;
    }

#ifndef NO_PKEY
```

And then you can leave the `init_cca_adapter_lock()` function untouched, except ifdefing out the attrs on AIX.